### DISCUSSION TITLE
Improvements to the navigation/route features

### DISCUSSION THREAD

**Discussion by SecularSteve** [FacilMap/facilmap#245] [Ideas]
For example, the route function is missing the option to use bus routes and/or combine different routes to get to your destination.

For example:

”Walk 50m to station > take the train > stop on another train > take the bus > walk another 20m > you have arrived"

I'd like for FacilMap to become an ideal navigation alternative. But it's all just a suggestion.

**Comment by cdauth** [comment]
This would be great. If you know of any OpenStreetMap-based service that can provide such route calculations, I would be happy to try to implement it.

**Comment by SecularSteve** [comment]
The best one that comes to mind is this one:

https://openrouteservice.org/
https://maps.openrouteservice.org/

There's also OsmAnd, but it's a paid service that's reliant on an "Offline Map" system.

Almost forgot (if you need it raw):

https://github.com/gboeing/osmnx

**Reply by cdauth** [reply]
Hm, but those don't support public transportation routing, do they?

**Comment by SecularSteve** [comment]
Found one that does support public transportation:

https://github.com/Bnyro/Transportr

I'll keep looking to see if I come across any more of them.
